modules/account-lib/src/coin/hbar/transactionBuilder.ts

```diff
@@ -99,6 +102,18 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     return this;
   }
 
+  /**
+   * Set an external transaction signature
+   *
+   * @param signature Hex encoded signature string
+   * @param keyPair The public key keypair that was used to create the signature
+   * @returns This transaction builder
+   */
+  signature(signature: string, keyPair: KeyPair): this {
+    this._signatures.push({ signature, keyPair });
```

Comment:
We should probably check the signature doesn't exist already for the keyPair pub. If it does, we can just replace it.